### ISSUE TITLE
pi4 64bit no dice

### ISSUE THREAD

**Issue by SpudGunMan** [KYDronePilot/hdfm#24]
sudo curl -sSL 'https://github.com/KYDronePilot/hdfm/releases/download/v2.0.1/hdfm-aarch64-unknown-linux-musl.tgz' | sudo tar xzv -C /usr/local/bin

uname -a
Linux pi4clock 5.15.84-v8+ #1613 SMP PREEMPT Thu Jan 5 12:03:08 GMT 2023 aarch64 GNU/Linux

$ hdfm 
bash: /usr/local/bin/hdfm: cannot execute binary file: Exec format error

**Comment by ferrellsl** [comment]
The first thing that I notice is that you didn't supply the frequency or channel number after typing hdfm.
Typical command line is:  hdfm 95.5 1
This will tune to freq. 95.5 on the station's 1st HD channel.

 

    On Thursday, February 16, 2023 at 04:21:20 PM MST, Kelly ***@***.***> wrote:  
 
 


sudo curl -sSL 'https://github.com/KYDronePilot/hdfm/releases/download/v2.0.1/hdfm-aarch64-unknown-linux-musl.tgz' | sudo tar xzv -C /usr/local/bin

uname -a
Linux pi4clock 5.15.84-v8+ #1613 SMP PREEMPT Thu Jan 5 12:03:08 GMT 2023 aarch64 GNU/Linux

$ hdfm
bash: /usr/local/bin/hdfm: cannot execute binary file: Exec format error

**Comment by SpudGunMan** [comment]
true I could run it with full options but it will have the same results, I thought this was related to missing libs or architecture  compile options?

$hdfm -v
bash: /usr/local/bin/hdfm: cannot execute binary file: Exec format error

**Comment by ferrellsl** [comment]
Did you build and install the nrsc5 binary first?  That's a required dependency listed on the github page.

 

    On Thursday, February 16, 2023 at 04:49:18 PM MST, Kelly ***@***.***> wrote:  
 
 


true I could run it with full options but it will have the same results, I thought this was related to missing libs or architecture compile options?

$hdfm -v
bash: /usr/local/bin/hdfm: cannot execute binary file: Exec format error

**Comment by SpudGunMan** [comment]
correct nrsc5 is built and there but when trying to run the HDFM the same error

**Comment by ferrellsl** [comment]
Verify that the nrsc5 binary is working by typing nrsc5 followed by a frequency and a channel number.  I.e. nrsc5 95.5 0 
 

Also, the 'v' argument for hdfm should be a capital V, not lowercase.
I suspect that the hdfm archive that you've downloaded is mangled so you may want to download and extract the binary again.  If that still doesn't work and if you've also installed Vulkan (another Linux dependency) and the binary still isn't working, then you may want to just install the binary from source. The author will release the sources to you if you send him a small donation and the compilation process is really easy.  On my system I just moved to the source directory and ran the following:  rustc main.rs
Another player that I recommend highly is here: https://github.com/jasonayu/nrsc5player

| 
| 
| 
|  |  |

 |

 |
| 
|  | 
GitHub - jasonayu/nrsc5player: NRSC5 Player is a Python-based graphical ...

NRSC5 Player is a Python-based graphical frontend for nrsc5 digital audio playback using an RTL-SDR dongle. Desi...
 |

 |

 |









    On Thursday, February 16, 2023 at 05:44:39 PM MST, Kelly ***@***.***> wrote:  
 
 


correct nrsc5 is built and there but when trying to run the HDFM the same error

**Comment by KYDronePilot** [comment]
Also try running with the 32 bit hdfm binary and see if that works

**Comment by SpudGunMan** [comment]
I will try just as a FYI
```
nrsc5 95.5 0
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.front
Detached kernel driver
Found Rafael Micro R820T tuner
Exact sample rate is: 1488375.071248 Hz
[R82XX] PLL not locked!

^C
$ hdfm 
bash: /usr/local/bin/hdfm: cannot execute binary file: Exec format error

```

**Comment by SpudGunMan** [comment]
I dont wanna pay for a project code that won't run ;)

**Comment by ferrellsl** [comment]
The nrsc5 binary is running fine so either there's problem with the hdfm archive or the hdfm binary in the archive is for another platform.  You still need to make sure that you have working audio on your Pi though.
Give one of these other nrsc5 players a try.  They're all very good but they're python based.
https://github.com/cmnybo/nrsc5-gui

https://github.com/markjfine/nrsc5-dui

https://github.com/jasonayu/nrsc5player







-- The quality of your thoughts will determine the quality of your life. 

    On Thursday, February 16, 2023 at 06:39:55 PM MST, Kelly ***@***.***> wrote:  
 
 


I will try just as a FYI
nrsc5 95.5 0
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.front
Detached kernel driver
Found Rafael Micro R820T tuner
Exact sample rate is: 1488375.071248 Hz
[R82XX] PLL not locked!

^C
$ hdfm 
bash: /usr/local/bin/hdfm: cannot execute binary file: Exec format error

**Comment by SpudGunMan** [comment]
that causes the binary errors? even a version? I am not eager to pay $20 if it won't load on the pi! I am kinda over it .. thank you for support but it seems a bad binary.

do you have this working on a pi4-64 aarch64?

**Comment by ferrellsl** [comment]
No, I have it running on Ubuntu 22.10 x86_64 and in Windows 11.



 

    On Thursday, February 16, 2023 at 07:18:29 PM MST, Kelly ***@***.***> wrote:  
 
 


that causes the binary errors? even a version? I am not eager to pay $20 if it won't load on the pi! I am kinda over it .. thank you for support but it seems a bad binary.

do you have this working on a pi4-64 aarch64?

**Comment by ferrellsl** [comment]
Well, I'm not the author so you'll have to contact him about the price and any payment.  I was just trying to be a helpful user.


 

    On Thursday, February 16, 2023 at 06:40:12 PM MST, Kelly ***@***.***> wrote:  
 
 


I dont wanna pay for a project code that won't run ;)
